lib/Connection.js

```diff
@@ -113,7 +113,11 @@ class Connection {
     const onError = err => {
       cleanListeners()
       this._openRequests--
-      callback(new ConnectionError(err.message), null)
+      let message = err.message
+      if (err.code === 'ECONNRESET') {
+        message += ` - Local: ${request.socket.localAddress}:${request.socket.localPort}, Remote: ${request.socket.remoteAddress}:${request.socket.remotePort}`
```

Comment:
TS definitions day `remoteAddress` and `remotePort` are optional. Should we fall back to a default value? `"unknown"`, for example

Reply:
Added a fallback, but as far as I know it should never happen, in case of ECONNRESET those properties should always be there.